### ISSUE TITLE
Ordering of `!important`

### ISSUE THREAD

**Issue by ChrisBAshton** [jakubpawlowicz/clean-css#833]
We use a grid library which makes use of the keyword `!important` quite a bit. In normal CSS, you can override a style by using a later style of the same or greater specificity, or an earlier style of a higher specificity. When using `!important`, the _last_ `!important` style is applied, regardless of specificity.

Therefore, the ordering of the CSS output is critical. I've found that `clean-css` will re-order our `!important` styles to prefer a smaller filesize output, even though this changes the intended style behaviour.

Example input:

```css
.gel-layout {
  direction: ltr;
  text-align: left;
  display: flex;
  flex-flow: row wrap;
  flex-grow: 1; }
  @media (min-width: 37.5em) {
    .gel-layout {
      margin-left: -16px; } }

.gel-layout--flush {
  margin-left: 0; }

.gel-1\/2 {
  width: 50% !important; }

.gel-1\/4 {
  width: 25% !important; }

@media (min-width: 37.5em) {
  .gel-1\/2\@m {
    width: 50% !important; }
  .gel-1\/4\@m {
    width: 25% !important; } }
```

Output:

```css
.gel-layout{direction:ltr;text-align:left;display:flex;flex-flow:row wrap;flex-grow:1}@media (min-width:37.5em){.gel-layout{margin-left:-16px}.gel-1\/2\@m{width:50%!important}.gel-1\/4\@m{width:25%!important}}.gel-layout--flush{margin-left:0}.gel-1\/2{width:50%!important}.gel-1\/4{width:25%!important}
```

...which leads to:

![screen shot 2016-11-10 at 16 57 21](https://cloud.githubusercontent.com/assets/5111927/20186169/d0d8d076-a766-11e6-8625-24b99511dea9.png)

^^ the intended behaviour is that the column is 50% wide until we reach a medium-sized screen, in which case it should switch to 25%. But because the media query comes before the `50%` declaration, and the 50% declaration is `!important`, the column is ALWAYS 50% width.

Feature suggestion:

* could you always respect the order of the original style declarations when `!important` is used?

PS: before you say "don't use `!important`", please read the library's justification:

> Utility classes like these width classes always need to win when it comes to specificity, otherwise they're not doing their job correctly. You should never need to override a utility class, if you do you're doing something wrong. E.g. if you'd used a class of gel-1/2 you would never want it to be anything other than 50%. If you want something other than 50% you should have used a different class.
> To help enforce this way of thinking all width utilities classes proactively carry the !important keyword to boost their specificity.
> More information: [The Importance of !important](http://csswizardry.com/2016/05/the-importance-of-important/)

Thanks.

**Comment by jakubpawlowicz** [comment]
First of all I'd never say "don't use `!important`". As long as it's a valid CSS it should be optimized properly.

That's totally a clean-css bug. In the meantime you can workaround it by using `--skip-restructuring` flag.

**Comment by ChrisBAshton** [comment]
Edit: before _anyone_ says "don't use `!important`". :-)

Thanks for the tip!

**Comment by jakubpawlowicz** [comment]
I actually was wrong regarding the flag - you should use `--skip-media-merging`.

**Comment by jakubpawlowicz** [comment]
It's fixed in 3.4.21.
